### PR TITLE
Return 404 on metadata missing resources not 400.

### PR DESCRIPTION
**Pull request by michael-mclawhorn**
I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

**Comment by cdeszaq** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- fili-core/src/main/java/com/yahoo/bard/webservice/web/apirequest/exceptions/MissingResourceApiRequestException.java  1
- fili-core/src/main/java/com/yahoo/bard/webservice/exception/FiliTablesExceptionHandler.java  1
- fili-core/src/main/java/com/yahoo/bard/webservice/exception/FiliMetricExceptionHandler.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Fili/fili/pullRequest?prid=5977528&bid=19600597)

File: fili-core/src/main/java/com/yahoo/bard/webservice/web/apirequest/SlicesApiRequestImpl.java [hunk 1]
```diff
@@ -143,11 +143,6 @@ public SlicesApiRequestImpl(
             PhysicalTableDictionary tableDictionary,
             UriInfo uriInfo
     ) throws BadApiRequestException {
-        if (tableDictionary.isEmpty()) {
-            String msg = EMPTY_DICTIONARY.logFormat("Slices cannot be found. Physical Table");
-            throw new BadApiRequestException(msg);
-        }
-
```
Comment (bharatmotwani) [inline fili-core/src/main/java/com/yahoo/bard/webservice/web/apirequest/SlicesApiRequestImpl.java:150]: How do we handle this case when tableDictionary is empty?  Looks like there is one more signature of generateSlices method.

**Comment by bharatmotwani** [comment]
It looks good to me. just had question on changes in SlicesApiRequestImpl

File: fili-core/src/main/java/com/yahoo/bard/webservice/web/apirequest/SlicesApiRequestImpl.java (hunk 1 above)
Comment (michael-mclawhorn) [reply fili-core/src/main/java/com/yahoo/bard/webservice/web/apirequest/SlicesApiRequestImpl.java:150]: If the dictionary is empty, we stream over the empty collection and return no slices without an error.
That's my proposed change over returning a 400 error.

**Comment by cdeszaq** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- fili-core/src/main/java/com/yahoo/bard/webservice/web/apirequest/exceptions/MissingResourceApiRequestException.java  1
- fili-core/src/main/java/com/yahoo/bard/webservice/exception/FiliTablesExceptionHandler.java  1
- fili-core/src/main/java/com/yahoo/bard/webservice/exception/FiliMetricExceptionHandler.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Fili/fili/pullRequest?prid=5977528&bid=19600597)

**Comment by cdeszaq** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- fili-core/src/main/java/com/yahoo/bard/webservice/exception/FiliTablesExceptionHandler.java  1
- fili-core/src/main/java/com/yahoo/bard/webservice/exception/FiliMetricExceptionHandler.java  1
- fili-core/src/main/java/com/yahoo/bard/webservice/web/apirequest/exceptions/MissingResourceApiRequestException.java  1
- fili-core/src/main/java/com/yahoo/bard/webservice/exception/FiliSlicesExceptionHandler.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Fili/fili/pullRequest?prid=5977528&bid=19600597)

**Comment by cdeszaq** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- fili-core/src/main/java/com/yahoo/bard/webservice/exception/FiliTablesExceptionHandler.java  1
- fili-core/src/main/java/com/yahoo/bard/webservice/web/apirequest/exceptions/MissingResourceApiRequestException.java  1
- fili-core/src/main/java/com/yahoo/bard/webservice/exception/FiliMetricExceptionHandler.java  1
- fili-core/src/main/java/com/yahoo/bard/webservice/exception/FiliSlicesExceptionHandler.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Fili/fili/pullRequest?prid=5977528&bid=19600597)

**Comment by cdeszaq** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- fili-core/src/main/java/com/yahoo/bard/webservice/exception/FiliMetricExceptionHandler.java  1
- fili-core/src/main/java/com/yahoo/bard/webservice/web/apirequest/exceptions/MissingResourceApiRequestException.java  1
- fili-core/src/main/java/com/yahoo/bard/webservice/exception/FiliTablesExceptionHandler.java  1
- fili-core/src/main/java/com/yahoo/bard/webservice/exception/FiliSlicesExceptionHandler.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Fili/fili/pullRequest?prid=5977528&bid=19600597)

**Comment by cdeszaq** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- fili-core/src/main/java/com/yahoo/bard/webservice/exception/FiliMetricExceptionHandler.java  1
- fili-core/src/main/java/com/yahoo/bard/webservice/exception/FiliSlicesExceptionHandler.java  1
- fili-core/src/main/java/com/yahoo/bard/webservice/web/apirequest/exceptions/MissingResourceApiRequestException.java  1
- fili-core/src/main/java/com/yahoo/bard/webservice/exception/FiliTablesExceptionHandler.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Fili/fili/pullRequest?prid=5977528&bid=19600597)

**Comment by cdeszaq** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- fili-core/src/main/java/com/yahoo/bard/webservice/exception/FiliSlicesExceptionHandler.java  1
- fili-core/src/main/java/com/yahoo/bard/webservice/exception/FiliTablesExceptionHandler.java  1
- fili-core/src/main/java/com/yahoo/bard/webservice/web/apirequest/exceptions/MissingResourceApiRequestException.java  1
- fili-core/src/main/java/com/yahoo/bard/webservice/exception/FiliMetricExceptionHandler.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Fili/fili/pullRequest?prid=5977528&bid=19600597)

**Review by bharatmotwani — APPROVED** [review]

**Pull request merged**